wordpress-mobile/WordPress-Android
# Issue/13328 13329 backup restore UI cycle 2

Parents #13328, #13329 and copy improvements #13916

This PR is part two of UI updates and includes
- Updated copy for Restore and Backup Download
- Updated view for status error
- Updated icon/copy for generic error views
- Split out building error views based on type (RemoteRequestFailure vs. GenericFailure)
- Introduced a new common item type `BACKUP_RESTORE_BULLET` and associated supporting classes `BulletState`, and `JetpackBackupRestoreBulletViewHolder`. Updated builders and adapter
- Added new icons for error views
- Caught and fixed an issue with setting error index for restore in `RestoreStep`

Merge Instructions
- Make sure #13898 has been merged
- Remove Not Ready for Merge Label
- Merge as normal

|restore status error|generic error|
|---|---|
|![status](https://user-images.githubusercontent.com/506707/106365515-2b281c00-6304-11eb-879d-5939cb93bbef.png)|![generic](https://user-images.githubusercontent.com/506707/106365523-31b69380-6304-11eb-9585-d7de6105de62.png)

|backup download status error|generic error|
|---|---|
|![status](https://user-images.githubusercontent.com/506707/106365525-354a1a80-6304-11eb-92ac-cec3350ff3ee.png)|![generic](https://user-images.githubusercontent.com/506707/106365532-3c712880-6304-11eb-8417-dcc3f5577bd1.png)

cc: @osullivanchris 
There weren't any details for backup download status (in #13916), so I winged it - same for the icon colors in status and generic error. Please provide any updates. Thanks. :)

To test:
- Forcing the remote request failure and generic failure states has to be done within the code manually
---------------------------------------------------
- In BackupDownloadViewModel, insert the following as the first two lines `fun buildDetails()`
- `transitionToError(BackupDownloadErrorTypes.RemoteRequestFailure)
        return`
- Build and run the app
- Turn on backup and restore feature flags
- Navigate to Activity Log
- Tap the more menu and tap the Download Backup option
- Note that the Status error view is shown
- Go back and change the two lines to: 
       `transitionToError(BackupDownloadErrorTypes.GenericFailure)
         return`
- Rebuild the app
- Navigate to Activity Log
- Tap the more menu and tap the Download Backup option
- Note that the Generic error view is shown
---------------------------------------------------
- In RestoreViewModel, insert the following as the first two lines `fun buildDetails()`
- `transitionToError(RestoreErrorTypes.RemoteRequestFailure)
        return`
- Build and run the app
- Turn on backup and restore feature flags
- Navigate to Activity Log
- Tap the more menu and tap the Restore to this point option
- Note that the Status error view is shown
- Go back and change the two lines to: 
       `transitionToError(RestoreErrorTypes.GenericFailure)
         return`
- Rebuild the app
- Navigate to Activity Log
- Tap the more menu and tap the Download Backup option
- Note that the Generic error view is shown




PR submission checklist:
